src/mem/Allocator.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_MEM_ALLOCATOR_H
+#define SQUID_SRC_MEM_ALLOCATOR_H
+
+#include "mem/forward.h"
+
+namespace Mem
+{
+
+/// An interface for memory allocators that deal with fixed-size objects.
+/// Allocators may optimize repeated de/allocations using memory pools.
+class Allocator
+{
+public:
+    Allocator(char const *aLabel) : label(aLabel) {}
+    virtual ~Allocator() {}
+
+    // TODO make this method const
+    /**
+     * fill the given object with statistical data about pool
+     * \returns Number of objects in use, ie. allocated.
+     */
+    virtual int getStats(MemPoolStats *, int accumulate = 0) = 0;
+
+    virtual MemPoolMeter const &getMeter() const = 0;
+
+    /// allocate one element from the pool
+    virtual void *alloc() = 0;
+
+    /// free a element allocated by Allocator::alloc()
+    virtual void freeOne(void *) = 0;
+
+    /// get a display label for objects in this pool
+    virtual char const *objectType() const { return label; }
+
+    /// the size (in bytes) of objects managed by this allocator
+    virtual size_t objectSize() const = 0;
+
+    /// the number of objects currently allocated
```

Comment:
The undefined term "allocated object" can be easily misinterpreted. We do not allocate objects. We provide memory for objects, but this method does not return the number of currently allocated memory blocks.

```suggestion
    /// the difference between the number of alloc() and freeOne() calls
```